great_expectations/experimental/datasources/sqlite_datasource.py

```diff
@@ -16,6 +107,45 @@ class SqliteDsn(pydantic.AnyUrl):
     host_required = False
 
 
+SqliteColumnSplitter = Union[
+    ColumnSplitter, ColumnSplitterHashedColumn, ColumnSplitterConvertedDateTime
+]
+
+
+class _SQLiteAssetMixin:
+    def add_splitter_hashed_column(
+        self: Self, column_name: str, hash_digits: int
+    ) -> Self:
+        return self._add_splitter(  # type: ignore[attr-defined]  # This is a mixin for a _SQLAsset
+            ColumnSplitterHashedColumn(
+                method_name="split_on_hashed_column",
+                column_name=column_name,
+                hash_digits=hash_digits,
+            )
+        )
+
+    def add_splitter_converted_datetime(
+        self: Self, column_name: str, date_format_string: str
+    ) -> Self:
+        return self._add_splitter(  # type: ignore[attr-defined]  # This is a mixin for a _SQLAsset
+            ColumnSplitterConvertedDateTime(
+                method_name="split_on_converted_datetime",
+                column_name=column_name,
+                date_format_string=date_format_string,
+            )
+        )
+
+
+class SqliteTableAsset(_SQLiteAssetMixin, SqlTableAsset):
+    type: Literal["sqlite_table"] = "sqlite_table"  # type: ignore[assignment]  # override superclass value
+    column_splitter: Optional[SqliteColumnSplitter] = None  # type: ignore[assignment]  # override superclass type
```

Comment:
What is driving going back to this mixin pattern? I'm not a fan of the `type: ignore`s in extended classes, but I guess its fine if we gain something somewhere else.

Reply:
Any new functionality we want to add, adding the new column splitters here, apply to both the table asset and the query asset. They don't share a common sqlite asset parent however, so I use this to prevent code duplication. You write the functionality once and mix it in to the engine specific table and query asset.

I just thought of this.  I could use dependency injection where I have a member variable in `_SQLAsset` called something like `engine_extensions`. I could override `__getattr__` or `__getattribute__` and delegate calls to `engine_extensions`.  I wouldn't need the sqlite datasset subclasses since I could inject in the sqlite engine extensions. However, we would need to update how we serialize/deserialize from config since we would need to know what `engine_extensions` to inject. Or I could still make subclasses and set the correct `engine_extensions` in the subclass. 

In terms of the type ignore, I agree I don't like it. Mypy doesn't do a good job with mixins since they are so dynamic and recommends using [protocols](https://mypy.readthedocs.io/en/latest/more_types.html#mixin-classes). I want to return `Self` here to be fluent and I don't think these interact well. There are a few other ways to avoid it:
* The dependency injection strategies above would avoid this. 
* I could define a property that calls super to avoid it (I don't think mypy yet checks the superclass hierarchy to see if it exists) or I could possibly declare an attribute. These seemed a little like tricking the type checker.
These all brought in forms of complication and I opted to ignore.